homeassistant/components/lidarr/__init__.py

```diff
@@ -40,31 +42,27 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         session=async_get_clientsession(hass, host_configuration.verify_ssl),
         request_timeout=60,
     )
-    coordinators: dict[str, LidarrDataUpdateCoordinator[Any]] = {
+    entry.runtime_data = {
         "disk_space": DiskSpaceDataUpdateCoordinator(hass, host_configuration, lidarr),
         "queue": QueueDataUpdateCoordinator(hass, host_configuration, lidarr),
         "status": StatusDataUpdateCoordinator(hass, host_configuration, lidarr),
         "wanted": WantedDataUpdateCoordinator(hass, host_configuration, lidarr),
     }
     # Temporary, until we add diagnostic entities
     _version = None
-    for coordinator in coordinators.values():
+    for coordinator in entry.runtime_data.values():
         await coordinator.async_config_entry_first_refresh()
         if isinstance(coordinator, StatusDataUpdateCoordinator):
             _version = coordinator.data
         coordinator.system_version = _version
-    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinators
     await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
```

Comment:
```suggestion
    entry.runtime_data = coordinators
    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
```

See https://github.com/home-assistant/core/pull/119551#discussion_r1637422489